dev/docker-compose.dev.confidentialbackend.yaml

```
@@ -0,0 +1,10 @@
+# docker-compose development override for confidentialbackend service
+version: "3.7"
+services:
+  confidentialbackend:
+    command: sh -c "flask run --host 0.0.0.0 --port ${PORT:-8009}"
```

Traefik uses the port advertised by a container's `EXPOSE` configuration to route HTTP traffic to it. You can view any ports exposed by a container as follows

```
$ docker inspect $(dc ps -q confidentialbackend) | grep -i expose -A2
            "ExposedPorts": {
                "8000/tcp": {}
            },
```

This should probably default to `8000`, [to match the port configured by `EXPOSE` in the docker image](https://github.com/uwcirg/helloworld-confidential-client-sof/blob/main/Dockerfile#L13-L16)

```suggestion
    command: sh -c "flask run --host 0.0.0.0 --port ${PORT:-8000}"
```

---

many thanks, @ivan-c .  makes perfect sense now - I had confused internal container ports with what docker-compose is exposing, thinking unique port numbers were necessary across services.